modules/services/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/SharingAnalytics.kt

```diff
@@ -6,10 +6,14 @@ import au.com.shiftyjelly.pocketcasts.analytics.AnalyticsTracker
 internal class SharingAnalytics(
     private val tracker: AnalyticsTracker,
 ) : SharingClient.Listener {
-    override fun onShare(request: SharingRequest) {
+    override fun onShare(
+        request: SharingRequest,
+        event: AnalyticsEvent,
+        eventProperties: Map<String, Any>,
+    ) {
```

Comment:
The idea behind `SharingAnalytics` class is that it handles all the analytics logic and it should be derived from the `SharingRequest`. Now the caller needs to consider which `event` should be used and if they need to pass in any additional `eventProperties`.

For example you can derive `event` from `SharingRequest` type as it is a sealed class.

Reply:
Got you. Let me give it another try.

Reply:
In 21e7588